src/core/runner.ts

```diff
@@ -150,24 +149,24 @@ export default class Runner {
     this.eventEmitter.on(e, cb);
   }
 
-  async runBeforeAllHook() {
+  async runBeforeAllHook(params: RunParamaters) {
```

Comment:
Can we rename this parameter runParams? It's confusing with both RunParams and Params existing

Reply:
I really have to change my keyboard :P, fixing it.

Reply:
Done

Reply:
Probbaly I misunderstood, if we change it to `runParams` in the args, do you want to pass them as `runParams` in the hooks as well?

Reply:
Okay I see the confusion here. I think we are on different pages. I had missed this change (https://github.com/elastic/synthetics/commit/cb01d24f6b073fd36b954ff7e16b10be4cb1e0ea#diff-fd47e6aef3e1ae139693b20fdc424b2a557e7214608e8f22e931b99e11d549bbR58) in the config refactor commit.

The question is, is the `environment` flag something that modifies a special key in `--suite-params` or is it a fully separate value passed on its own? Given that they are passed as separate flags I don't think they should be mixed into the same map. I think we should expose it as a fully separate variable as a sibling alongside params.

So, we'd have a signature like:

`beforeAll({params, env} => {})` Instead of what we have today. What do you think?

Reply:
We agreed on splitting param and env offline. 

will make the changes here and ask for review.

Reply:
Done.